plugins/module_utils/prism/clusters.py

```diff
@@ -5,13 +5,104 @@
+            "supported_information_verbosity": self._build_spec_supported_information_verbosity,
+            "redundancy_factor": self._build_spec_redundancy_factor,
+            "network": self._build_spec_network,
+        }
 
+    def _get_default_spec(self):
+        return deepcopy(
+            {
+                "api_version": "3.1.0",
+                "metadata": {"kind": "cluster", "spec_version": 2},
+                "spec": {
+                    "name": None,
+                    "resources": {
+                        "config": {
+                            "authorized_public_key_list": []
+                        }
+                    }
+                }
+            }
+        )
+
+    def _build_spec_redundancy_factor(self, payload, redundancy_factor):
+        if redundancy_factor is not None:
+            payload["spec"]["resources"]["config"]["redundancy_factor"] = redundancy_factor
+        return payload, None
+
+    def _build_spec_supported_information_verbosity(self, payload, verbosity):
+        if verbosity is not None:
+            payload["spec"]["resources"]["config"]["supported_information_verbosity"] = verbosity
+        return payload, None
+
+    def _build_spec_timezone(self, payload, timezone):
+        if timezone is not None:
+            payload["spec"]["resources"]["config"]["timezone"] = timezone
+        return payload, None
+
+    def _build_spec_public_keys(self, payload, public_keys):
+        if public_keys is not None:
+            payload["spec"]["resources"]["config"]["authorized_public_key_list"] = public_keys
+        return payload, None
+
+    def _build_spec_network(self, payload, network):
+        single_entries = ["external_ip", "fully_qualified_domain_name", "external_data_services",
+                          "external_subnet", "internal_subnet", "masquerading_ip", "masquerading_port", "domain_server",
+                          "nfs_subnet_whitelist", "name_server_ip_list", "ntp_server_ip_list",
+                          "http_proxy_list", "http_proxy_whitelist", "default_vswitch_config"]
+
+        for single_entry in single_entries:
+            if single_entry in network:
+                payload["spec"]["resources"]["network"][single_entry] = network[single_entry]
+
+        # special snowflake with keyword option collision.  smtp server has an argument of "type"
+        # which is a keyword in the ansible argument spec.  this will remap the value back to type in the
+        # spec file is specified.  this will change server_type back to type.  the PLAIN default is
+        # used here if nothing is found in the config or existing spec file
+        if "smtp_server" in network:
+            if "server_type" in network["smtp_server"]:
+                network["smtp_server"]["type"] = network["smtp_server"]["server_type"]
+                del network["server_type"]
+            elif "type" in payload["spec"]["resources"]["network"]["smtp_server"]:
```

Comment:
```suggestion
            elif "type" in (payload["spec"]["resources"]["network"]).get("smtp_server"):
```
Get API response might not have stmp server entry. Failed for me in local in this case.